src/components/views/rooms/MessageComposerInput.js

```diff
@@ -336,6 +343,14 @@ export default class MessageComposerInput extends React.Component {
                 this.onFinishedTyping();
             }
 
+            // Record the editor state for this room so that it can be retrieved after
+            // switching to another room and back
+            dis.dispatch({
+                action: 'editor_state',
+                room_id: this.props.room.roomId,
+                editor_state: state.editorState,
+            });
```

Comment:
If I've understood flux correctly, this isn't really an 'action' as such, you're basically just telling the store to store a thing which is not really fluxy. This may be a symptom of the editor state not having deeper flux integration in the first place though, and I'm not about to propose rewriting all that.